llvm/lib/Target/RISCV/RISCVTargetTransformInfo.cpp

```diff
@@ -1134,16 +1134,13 @@ InstructionCost RISCVTTIImpl::getCastInstrCost(unsigned Opcode, Type *Dst,
       MVT ElementVT = MVT::getIntegerVT(SrcEltSize >> 1);
       MVT VecVT = DstLT.second.changeVectorElementType(ElementVT);
       Cost += getRISCVInstructionCost(FNCVT, VecVT, CostKind);
-    }
-    if ((SrcEltSize >> 1) > DstEltSize) {
-      // For mask type, we use:
-      // vand.vi v8, v9, 1
-      // vmsne.vi v0, v8, 0
-      VectorType *VecTy =
-          VectorType::get(IntegerType::get(Dst->getContext(), SrcEltSize >> 1),
-                          cast<VectorType>(Dst)->getElementCount());
-      Cost +=
-          getCastInstrCost(Instruction::Trunc, Dst, VecTy, CCH, CostKind, I);
+      if ((SrcEltSize >> 1) > DstEltSize) {
+        VectorType *VecTy = VectorType::get(
+            IntegerType::get(Dst->getContext(), SrcEltSize >> 1),
+            cast<VectorType>(Dst)->getElementCount());
```

Comment:
Can we use `getTypeForEVT` so we reuse VecVT? Otherwise I don't think we use the legalized type

Reply:
Cool, thanks for catching that!